### PR TITLE
Smallpatches

### PR DESCRIPTION
**Pull request by InfusOnWoW**

### CODE REVIEW

File: WeakAurasOptions/RegionOptions/Icon.lua [hunk 1]
```diff
@@ -153,12 +159,45 @@ local function createOptions(id, data)
...
+      type = "range",
+      control = "WeakAurasSpinBox",
+      width = WeakAuras.normalWidth,
+      name = L["Texture Y Offset"],
+      order = 7.07,
+      min = -1,
+      max = 1,
+      bigStep = 0.1,
+      hidden = function()
+        hiddenIconExtra()
```
Comment (mrbuds) [inline WeakAurasOptions/RegionOptions/Icon.lua:185]: It should be `hidden = hiddenIconExtra`  for texXOffset and texYOffset

File: WeakAuras/RegionTypes/Icon.lua [hunk 2]
```diff
@@ -139,8 +141,11 @@ local function GetTexCoord(region, texWidth, aspectRatio)
   local xRatio = aspectRatio < 1 and aspectRatio or 1;
   local yRatio = aspectRatio > 1 and 1 / aspectRatio or 1;
   for i, coord in ipairs(region.currentCoord) do
-    local aspectRatio = (i % 2 == 1) and xRatio or yRatio;
-    region.currentCoord[i] = (coord - 0.5) * texWidth * aspectRatio + 0.5;
+    if(i % 2 == 1) then
+      region.currentCoord[i] = (coord - 0.5) * texWidth * xRatio + 0.5 + xOffset;
```
Comment (mrbuds) [inline WeakAuras/RegionTypes/Icon.lua:143]: I think it should be `- xOffset` here, negative offset move icon to the right

**Comment by mrbuds** [comment]
It seems Sparks no longer work
```
!WA:2!DrvZUTTrqytqGeuH4ahLwJMIc0a7g3FasrHaQpuGIarzPiNyl5srh7ueaPLChkU1u7YU7sjlJEPcff5SYdqb05Esh6dG7laHAFc8JGFbC3DjJLt0bHDg(nZoZ38nRvJYdkJlJFDLzIee)KJiyz0hnJeWODyP8a4txzbkvgX4TtKegvuIpUDyOaKwZr0aL)dyeQ0Vw9wE1DxyYHltI0yTsU37yVpddovp0R9L(OGt6ZzPuCnwmJ)mllRdS)2V57Mi5K(9bU4UBXlo(3oYXjGtkLitkNkGUiFbJ73eqX6817crQpmeOYoPHHKtN3Tw1oED74v117ILWnqDn4CQHeYRJ6aoOIYTZb13BVjIeioExSO0eDK6RSNlfnaeLYURFsmAmWNHbF1f5PQjEZ67DqJd3BAkTOwlLCpuGKme8YT136FTIHm7gyAuB1Vlv8fWPO4xOAufTmEkIsgyyOTDfsex2iBvhkJcNdib0rYbAFz0TxGt5guD1mcxaQ0Ift0y0Lt2ModqeQk2SnZ(8BN9OSTY(ISn9djuIi69DpXhXZ5EBllBvxjqsD6HO5HCvlRUvKezFUzaMpWFP1mfR5c91YGVF9SvxOmxojlCeY4qUJx)HjLLWPYUIieMn648Syp14t)3gpIMS2nqCTwW(ICVqCOrDL9bj334rPezAMkWitDQQmV08b1meWgN)2ICz5lluPNB((pNkKKWXz3AgxlgbpLtNwTBvpBvUQO1yMzaUCmLxNHmQmzTgCYzp8htrynP8qpV3PXkUj1IIX5ighFehLm5OIdlUor6XKtl1GTOTVXcKsbw1955f7qIG4hd9wgyhYzWdkAScE0k5bxtKkmCfT0LwCqn1Jas)i5N5eZq4gocv8R7oinwsk5Ou5bRN1QKBqmsi0N8LOy1UG(4fMbUJYeRvVvQUZojRP2gQI1uiGb8(OtJsUFFoctuX0MR)3ilN1ST7U)u7wEv3B6BvxvuuPACcuKQHEArm9C0BerVFEj0OzCJ6YSDHuYrvAUPc8yl(PfD)uZsLGGbx3DFAtpFLziPFPni4kTGrZnr10qc)axZrPCyYBlGlZzYgAHU(U87uZTE9w5xLxo4KVCx9AAikaEL(fdcTVdI)Qd39XlTEChZZBMW6ntYcgMVr)Xw)XSIkGGXa1Tv9xu315mgBGLlkojcz7oslw)97ujLGN)l4rT33)5OK0IxHRDTkKtO6KcrC9llQfVTZ2OKJALwwk7RknNTK(ZEXm9dcedQpzl3Gii4KTZUJ9KHiorZ)tHbjSrahWUdrXPGnpicr7dk0z)6ZA(K)5nV5pV6QRSB(K)TtNERSYk2ts4m1Oqoo7RNtO5Ilv6lnD5Q5TklQSdmS8W)74))
```
Spark doesn't move with progress, neither text anchored to spark

**Comment by InfusOnWoW** [comment]
> Spark doesn't move with progress, neither text anchored to spark
Should be fixed now.

File: WeakAuras/Init.lua [hunk 3]
```diff
@@ -50,6 +50,7 @@ Private.frames = {}
 --- @field glow_types table<string, string>
 --- @field inverse_point_types table<string, string>
 --- @field IsCLEUSubevent fun(subevent: string): boolean
+--- @field IsDragonRiding fun(): boolean
```
Comment (Stanzilla) [inline WeakAuras/Init.lua:53]: Dragonriding

File: WeakAuras/Prototypes.lua [hunk 4]
```diff
@@ -1138,6 +1138,17 @@ Private.load_prototype = {
       hidden = not WeakAuras.IsWrathOrRetail(),
       events = {"VEHICLE_UPDATE", "UNIT_ENTERED_VEHICLE", "UNIT_EXITED_VEHICLE", "UPDATE_OVERRIDE_ACTIONBAR", "UPDATE_VEHICLE_ACTIONBAR"}
     },
+    {
+      name = "dragonRiding",
```
Comment (Stanzilla) [inline WeakAuras/Prototypes.lua:1142]: dragonriding

File: WeakAuras/WeakAuras.toc [hunk 5]
```diff
@@ -52,6 +52,7 @@ AuraWarnings.lua
 AuraEnvironment.lua
 AuraEnvironmentWrappedSystems.lua
 DebugLog.lua
+DragonRiding.lua
```
Comment (Stanzilla) [inline WeakAuras/WeakAuras.toc:55]: Dragonriding

**Review by Stanzilla — CHANGES_REQUESTED** [review]

**Comment by Stanzilla** [comment]
I mentioned it in the ticket but not here, there is supposed to be a DMUI_DRAGONRIDING_SHOW event

**Comment by mrbuds** [comment]
This error show on login on wotlk

Message: Interface\AddOns\WeakAuras\WeakAuras.lua:1561: attempt to call field 'IsDragonriding' (a nil value)
[string "@Interface\AddOns\WeakAuras\WeakAuras.lua"]:1561: in function <Interface\AddOns\WeakAuras\WeakAuras.lua:1524>
[string "@Interface\AddOns\WeakAuras\WeakAuras.lua"]:1870: in function `Resume'
[string "@Interface\AddOns\WeakAuras\WeakAuras.lua"]:1159: in function <Interface\AddOns\WeakAuras\WeakAuras.lua:1121>

**Review by Stanzilla — CHANGES_REQUESTED** [review]
WeakAurasOptions/RegionOptions/Icon.lua has a few semicolons added that should be removed, otherwise lgtm

**Comment by InfusOnWoW** [comment]
@Stanzilla those are due to starting with a revert of these files and then fixing them up again. I'm not going to bother on removing semicolons from that..

**Pull request merged**